src/platform/console/console.cpp

```diff
@@ -16,14 +16,16 @@
  */
 
 #include <multipass/console.h>
+#include <multipass/terminal.h>
 
 #include "unix_console.h"
+#include "unix_terminal.h"
 
 namespace mp = multipass;
 
-mp::Console::UPtr mp::Console::make_console(ssh_channel channel)
+mp::Console::UPtr mp::Console::make_console(ssh_channel channel, Terminal* term)
 {
-    return std::make_unique<UnixConsole>(channel);
+    return std::make_unique<UnixConsole>(channel, static_cast<UnixTerminal*>(term));
```

Comment:
For my instruction, how does this platform stuff work? Are the contents of the `/src/platform` directory somehow different in other platforms? And is that why this `static_cast` is expected to succeed?

Reply:
Yep we add extra bits for other platforms. The #ifdefs will prevent accidental mixing of platform components, which is why that static_cast will succeed - on a unix-system, only UnixTerminal will exist.